File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/effect/UXLightingEffect.php
<?php
namespace php\gui\effect;

use php\gui\paint\UXColor;

/**
 * Class UXLightingEffect
 * @package php\gui\effect
 * @packages gui, javafx
 */
class UXLightingEffect extends UXEffect
{
    /**
     * @var double 0.0 to 2.0
     */
    public $diffuseConstant = 1;

    /**
     * @var double 0.0 to 2.0
     */
    public $specularConstant = 0.3;

    /**
     * @var float 0.0 to 40.0
     */
    public $specularExponent = 20.0;

    /**
     * @var float 0 to 10
     */
    public $surfaceScale = 1.5;

    /**
     * @var UXEffect
     */
    public $contentInput = null;

    /**
     * @var UXEffect
     */
    public $bumpInput = null;

    /**
     * @param double $threshold (optional)
     */
    public function __construct($threshold)
    {
    }
}

File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/effect/UXLightingEffect.java
package org.develnext.jphp.ext.javafx.classes.effect;

import javafx.scene.effect.Bloom;
import javafx.scene.effect.Effect;
import javafx.scene.effect.Light;
import javafx.scene.effect.Lighting;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "effect\\UXLightingEffect")
public class UXLightingEffect extends UXEffect<Lighting> {
    interface WrappedInterface {
        @Property double diffuseConstant();
        @Property double specularConstant();
        @Property double specularExponent();
        @Property double surfaceScale();
        @Property Effect contentInput();
        @Property Effect bumpInput();
    }

    public UXLightingEffect(Environment env, Lighting wrappedObject) {
        super(env, wrappedObject);
    }

    public UXLightingEffect(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new Lighting();
    }
}
